internal/tfdiags/diagnostics.go

```diff
@@ -101,6 +101,23 @@ func (diags Diagnostics) Append(new ...interface{}) Diagnostics {
 	return diags
 }
 
+func (diags Diagnostics) StrictDeduplicateMerge(other Diagnostics) Diagnostics {
```

Comment:
Can we say that diags are duplicates if there is no source range set in them?

Those diags (same description and severity) could reference different parts of configuration and should not be treated as duplicates.

Reply:
If everything else is identical, and there is no source range, I don't see how hiding one of two duplicate errors would cause any difficulty.